### PR TITLE
Bot refactoring for easier use and updating

### PR DESCRIPTION
**Pull request by StevenBucher98**
# PR Summary

Trying to update the github resource management bot so we can see the behaviors more clearly and segmented out, WIP as I mainly want the bot itself to check for syntax issues

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: .github/policies/IssueManagement.IssueFeedbackForm.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+disabled: false
+where:
+configuration:
+  resourceManagementConfiguration:
+    eventResponderTasks:
+      - description: Automatically commenting for feedback when an issue is closed,
+        triggerOnOwnActions: true
+        if: 
+          - isAction:
+            action: Closed
```
Comment (markphip) [inline .github/policies/IssueManagement.IssueFeedbackForm.yml:15]: Indentation needs to be added here and the next item

File: .github/policies/IssueManagement.ResolveStale.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      - isIssue
+      - isOpen
+      - isNotLabeledWith:
+          label: KeepOpen
+      - isNotLabeledWith:
+          label: Needs-Triage
+      - isNotLabeledWith:
+          label: Resolution-No Activity
+      - isNotLabeledWith:
+          label: Issue-Meta
```
Comment (StevenBucher98) [inline .github/policies/IssueManagement.ResolveStale.yml:40]: add labels, exclude- "WG-Needs Review" and "Review-Commitee"

File: .github/policies/IssueManagement.ResolveStale.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      - noActivitySince:
+          days: 180
+      actions:
+      - addLabel:
+          label: Resolution-No Activity
+      - addReply:
+          reply: "This issue has not had any activity in 6 months, if there is no further activity in seven days, the issue will be closed automatically.\n\nActivity in this case refers to comments on the issue. If the issue is closed and you are the author, you can re-open the issue using the button below. Please add more information to be considered during retriage. If you are not the author but the issue is impacting you after it has been closed, please submit a new issue with updated details and a link to this issue and the original."
```
Comment (StevenBucher98) [inline .github/policies/IssueManagement.ResolveStale.yml:47]: "activity only refers to ..."

Comment (StevenBucher98) [reply .github/policies/IssueManagement.ResolveStale.yml:47]: emphasis 7 days

Comment (JamesWTruher) [inline .github/policies/IssueManagement.ResolveStale.yml:47]:
```suggestion
          reply: "This issue has not had any activity in 6 months, if there is no further activity in seven days, the issue will be closed automatically.\n\nActivity in this case refers only to comments on the issue. If the issue is closed and you are the author, you can re-open the issue using the button below. Please add more information to be considered during retriage. If you are not the author but the issue is impacting you after it has been closed, please submit a new issue with updated details and a link to this issue and the original."
```

File: .github/policies/IssueManagement.ResolveStale.yml (hunk 2 above)
Comment (daxian-dbw) [inline .github/policies/IssueManagement.ResolveStale.yml:40]: And `up-for-grab`?

File: .github/policies/IssueManagement.CloseResolutions.yml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+name: GitOps.PullRequestIssueManagement
+description: Closing issues with Resolution* 
+owner:
+resource: repository
+disabled: false
+where:
+configuration:
+  resourceManagementConfiguration:
+    scheduledSearches:
+    - description: Close if marked as Resolution-Declined after one day of no activity
```
Comment (daxian-dbw) [inline .github/policies/IssueManagement.CloseResolutions.yml:11]: I would be more readable if you can insert 1 empty line before each `- description: xxxx` (except for the 1st one), so it's easy to see where one ends and where one starts.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: .github/policies/PRManagement.yml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      then:
+      - removeLabel:
+          label: Review - Needed
+    - description: If inPRLabel then label in PR
+      if:
+      - payloadType: Pull_Request
+      then:
+      - inPrLabel:
+          label: In-PR
+    - description: Clean email reply if issue comment - NOTE may be broke
```
Comment (StevenBucher98) [inline .github/policies/PRManagement.yml:211]: remove

File: .github/policies/IssueManagement.ResolveStale.yml (hunk 2 above)
Comment (StevenBucher98) [reply .github/policies/IssueManagement.ResolveStale.yml:40]: the committee decided to have up for grab labels also close after 6 months if there is no outside contributions

**Comment by daxian-dbw** [comment]
@StevenBucher98 You need to resolve the conflicts.

**Review by daxian-dbw — APPROVED** [review]

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Review by kilasuit — COMMENTED** [review]
I started to review this but due to other things GitHub lost my review, which actually may be for the best as was much longer than this one will be & let us at least get this in and then can be discussed about further adaptions & improvements.

In short @StevenBucher98 I think this PR was good & needed, especially the splitting up of things into smaller files, though perhaps some language in messages and the time frames for closures could be improved to make the experience seem friendlier & account for both the issue/discussion/PR author/collaborators/maintainers having time off like weekends, non-working days, illness & holidays being or collaborators/maintainers unsure as to where to initially triage to as the team doesn't (I don't think) currently or intend to provide a 24x7x365 support experience, but should detail via a response that links to a support policy so that when new issues are raised people can read and get an instant understanding of what should come next & if needs what escalation processes are in place, both via being community/team maintainers/collaborators. 

I think we pretty much have almost all of this in place already, but I think but it may just need to connect some missing dots and add an auto response on issue/PR/discussion creation that can link to either a doc or blog post detailing how the team expects to be deal with interactions going forward to better formalise things and make expectations clear, this can then be used as a starting blueprint for interaction across all repos/services of the PowerShell team and I hope should make experiences better for all involved going forward & probs better to see if we can try find some time to discuss more in the new year.

**Pull request merged**